src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,758 @@
+      regionBasedKernelApplication< EXEC_POLICY,
+                                    constitutive::NullModel,
+                                    CellElementSubRegion >( mesh,
+                                                            regionNames,
+                                                            getDiscretizationName(),
+                                                            "",
+                                                            kernelFactory );
+
+
+    addSourceToRightHandSide( cycleNumber, rhsx, rhsy, rhsz );
+
+
+
+    real64 dt2 = dt*dt;
+    forAll< serialPolicy >( nodeManager.size(), [=] ( localIndex const a )
+    {
+      if( freeSurfaceNodeIndicator[a]!=1 )
+      {
+        ux_np1[a] = ux_n[a];
+        ux_np1[a] *= 2.0*mass[a];
+        ux_np1[a] -= (mass[a]-0.5*dt*dampingx[a])*ux_nm1[a];
+        ux_np1[a] += dt2*(rhsx[a]-stiffnessVectorx[a]);
+        ux_np1[a] /= mass[a]+0.5*dt*dampingx[a];
+        uy_np1[a] = uy_n[a];
+        uy_np1[a] *= 2.0*mass[a];
+        uy_np1[a] -= (mass[a]-0.5*dt*dampingy[a])*uy_nm1[a];
+        uy_np1[a] += dt2*(rhsy[a]-stiffnessVectory[a]);
+        uy_np1[a] /= mass[a]+0.5*dt*dampingy[a];
+        uz_np1[a] = uz_n[a];
+        uz_np1[a] *= 2.0*mass[a];
+        uz_np1[a] -= (mass[a]-0.5*dt*dampingz[a])*uz_nm1[a];
+        uz_np1[a] += dt2*(rhsz[a]-stiffnessVectorz[a]);
+        uz_np1[a] /= mass[a]+0.5*dt*dampingz[a];
+      }
+    } );
+
+    /// synchronize pressure fields
+    FieldIdentifiers fieldsToBeSync;
+    fieldsToBeSync.addFields( FieldLocation::Node, { extrinsicMeshData::Displacementx_np1::key(), extrinsicMeshData::Displacementy_np1::key(), extrinsicMeshData::Displacementz_np1::key() } );
+
+    CommunicationTools & syncFields = CommunicationTools::getInstance();
+    syncFields.synchronizeFields( fieldsToBeSync,
+                                  domain.getMeshBody( 0 ).getMeshLevel( 0 ),
+                                  domain.getNeighbors(),
+                                  true );
+
+    // compute the seismic traces since last step.
+    arrayView2d< real64 > const uxReceivers   = m_displacementxNp1AtReceivers.toView();
+    arrayView2d< real64 > const uyReceivers   = m_displacementyNp1AtReceivers.toView();
+    arrayView2d< real64 > const uzReceivers   = m_displacementzNp1AtReceivers.toView();
+
+    computeAllSeismoTraces( time_n, dt, ux_np1, ux_n, uxReceivers );
+    computeAllSeismoTraces( time_n, dt, uy_np1, uy_n, uyReceivers );
+    computeAllSeismoTraces( time_n, dt, uz_np1, uz_n, uzReceivers );
+
+    forAll< EXEC_POLICY >( nodeManager.size(), [=] GEOSX_HOST_DEVICE ( localIndex const a )
+    {
+      ux_nm1[a] = ux_n[a];
+      uy_nm1[a] = uy_n[a];
+      uz_nm1[a] = uz_n[a];
+      ux_n[a] = ux_np1[a];
+      uy_n[a] = uy_np1[a];
+      uz_n[a] = uz_np1[a];
+
+      stiffnessVectorx[a] = 0.0;
+      stiffnessVectory[a] = 0.0;
+      stiffnessVectorz[a] = 0.0;
+      rhsx[a] = 0.0;
+      rhsy[a] = 0.0;
+      rhsz[a] = 0.0;
+    } );
```

Comment:
```suggestion
    } );
    for( real64 timeSeismo;
     (timeSeismo = m_dtSeismoTrace*m_indexSeismoTrace) <= (time_n + epsilonLoc) && m_indexSeismoTrace < m_nsamplesSeismoTrace;
           m_indexSeismoTrace++ )
     {
        /// do nothing 
     }
```
See my comment below to why this is needed.